### PR TITLE
For each week, for each fine area station concatenate the weather data

### LINKED ISSUE #38: Combine all weekly weather data in one dataframe
## Description
The goal of this ticket is to group the data by week and by fine area of interest. 

Right now our weather data is in the following structure:
```
data 
   weather
      year 
         week
            network
               station weather data
```

Right now our station data is in the following structure:
```
data 
   weather-stations
      kelowna
         fine area stations
```
   
Our final data should be in the following structure:
```
data 
   final-weather
      year 
         week
            fine-area-weather-data (weather data for all networks and stations in this fine area)
```

## Sample output 
```
data 
   final-weather
      2017 
         week-1
            fine-area-1-weather.csv
            fine-area-2-weather.csv
            ...
            fine-area-144-weather.csv
         week-2
            fine-area-1-weather.csv
            fine-area-2-weather.csv
            ...
            fine-area-144-weather.csv
         ...
         week-52
            fine-area-1-weather.csv
            fine-area-2-weather.csv
            ...
            fine-area-144-weather.csv
      2018
      ...
      2022
```

Each `fina-area-#-weather.csv` should look something like this (a list of all weather observations from stations in the fine area of interest): 
```
ONE_DAY_PRECIPITATION, ONE_DAY_RAIN, MIN_TEMP, ONE_DAY_SNOW, time, SNOW_ON_THE_GROUND, MAX_TEMP
0.0, 0.0, -11.0, 0.0, 2017-01-04 00:00:00, 18.0, -4.0
18.0, 0.0, -13.0, 18.0, 2017-01-05 00:00:00, 18.0, -4.0
8.0, 0.0, -9.0, 8.0, 2017-01-06 00:00:00, 25.0, 0.0
0.0, 0.0, -7.0, 0.0, 2017-01-09 00:00:00, 33.0, -2.0
5.0, 0.0, -20.0, 5.0, 2017-01-12 00:00:00, 0.0, -7.0
8.5, 6.5, -11.0, 2.0, 2017-01-13 00:00:00, 35.0, 3.0
34.0, 34.0, 1.5, 0.0, 2017-01-16 00:00:00, 22.0, 6.0
34.0, 34.0, -0.5, 0.0, 2017-01-17 00:00:00, 20.0, 3.0
13.0, 13.0, 0.0, 0.0, 2017-01-18 00:00:00, 18.0, 3.0
```

 
## Relation to your old task
This is similar to your work in `gen_yearly_data_script.ipynb`. 

In that task, you took in a `filtered list of stations` and grouped all the weather data in those stations for each year. 

For this task, your `filterest list of stations` is your `fine area stations` that you just created in your last task.

You want to iterate on each of your `fine area stations` lists and group all the weather in those stations for each week.

## Flow Chart 
Image

### PR DESCRIPTION
**Pull request by Ari1029**

### CODE REVIEW

**Comment by Ari1029** [comment]
Screen Shot 2024-02-14 at 9 42 31 PM
Example concatenated file.

File: final_weather_concatentation_script.ipynb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (rchang0501) [inline final_weather_concatentation_script.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.                    combined_df.fillna(-999, inplace = True)
</pre><p>add comment explaining why we're filling the null columns with this value</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WAT-ai/wildfire/pull/53/discussion/#comment-1493399589'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rchang0501) [inline final_weather_concatentation_script.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #24.                    if not os.path.exists(f'./data/final-weather/{curr_year}/week_{week}'):
</pre><p>change the file path to <code>f'./data/final-weather/{curr_year}/week_{week}'</code></p><p> </p><p>apply for the bottom two as well </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WAT-ai/wildfire/pull/53/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by rchang0501 — APPROVED** [review]
looks good! just the one comment

remember to `squash and merge`!

**Pull request merged**